src/lib/alloc.c

```diff
@@ -268,7 +269,7 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
 		map->first_free += count;
 
 	/* update each block */
-	for (current = start; current < count; current++) {
+	for (i = 0, current = start; i < count; i++, current++) {
```

Comment:
What about this:

```diff
diff --git a/src/lib/alloc.c b/src/lib/alloc.c
index 61d180b1..ece121b7 100644
--- a/src/lib/alloc.c
+++ b/src/lib/alloc.c
@@ -267,9 +267,9 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
        if (map->first_free == start)
                map->first_free += count;

-       /* update each block */
-       for (current = start; current < count; current++) {
-               hdr = &map->block[current];
+       /* update each allocated block */
+       for (current = 0; current < count; current++, start++) {
+               hdr = &map->block[start];
                hdr->used = 1;
        }


```

Reply:
I would rather like something like:
'''
index 61d180b..e64e26c 100644
--- a/src/lib/alloc.c
+++ b/src/lib/alloc.c
@@ -232,6 +232,7 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
        unsigned int current;
        unsigned int count = bytes / map->block_size;
        unsigned int remaining = 0;
+       unsigned int stop;

        if (bytes % map->block_size)
                count++;
@@ -268,7 +269,8 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
                map->first_free += count;

        /* update each block */
-       for (current = start; current < count; current++) {
+       stop = start + count;
+       for (current = start; current < stop; current++) {
                hdr = &map->block[current];
                hdr->used = 1;
        }
'''
since current refers to block id. What is your opinion?

Reply:
Yes, my version overwrites `current`. Anyway after update your patch looks good to me now.